client/src/routes/settings/+page.svelte

```diff
@@ -194,9 +257,107 @@
     </Dialog>
   {/if}
 
-  <Button on:click={() => (showDeletionPrompt = true)}>
-    <Label>Delete Account</Label>
-  </Button>
+  <p id="email">Email: {$user?.email == null ? "Anonymous User" : $user.email}</p>
+  <h3>Link Provider Options</h3>
+
+  <div class="signin-buttons">
+    <ProviderButtons
+      on:google-clicked={linkGoogleAccount}
+      on:microsoft-clicked={linkMicrosoftAccount}
+      hideGoogle={hasGoogleProvider($user)}
+      hideMicrosoft={hasMicrosoftProvider($user)}
+    />
+  </div>
+  <div>
+    <Button on:click={() => (showOptions = true)}>
+      <Label>Set Password</Label>
+    </Button>
+  </div>
+  <Dialog bind:open={showOptions} aria-labelledby="set-password-title" aria-describedby="link-options-content">
+    <Title id="set-password-title">Set Password</Title>
+    <Content id="set-password-content">
+      <Textfield
+        name="password"
+        label="Password"
+        type={showPassword ? "text" : "password"}
+        bind:value={password}
+        required
+      >
+        <IconButton
+          type="button"
+          on:click={(event) => event.preventDefault()}
+          slot="trailingIcon"
+          toggle
+          bind:pressed={showPassword}
+        >
+          <Icon class="material-icons" on>visibility</Icon>
+          <Icon class="material-icons">visibility_off</Icon>
+        </IconButton>
+      </Textfield>
+      <div>
+        <Button on:click={linkPassword}>
+          <Label>Set Password</Label>
+        </Button>
+      </div>
+      {#if linkPass}
+        <p>Password Sucessfully Set</p>
+      {:else if passErr !== ""}
+        <p class="error">{outputErrMsg(passErr)}</p>
+      {/if}
+    </Content>
+  </Dialog>
+  <div>
+    <Button
+      on:click={() => {
+        logOut();
+        goto("/");
```

Comment:
this should do a page reload like it does for the sign out button in the account dropdown